configure.ac

```diff
@@ -142,25 +142,26 @@ AS_IF([test "x$have_jsoncpp" = "xyes"], [
 ])
 
 AC_ARG_WITH([openssl], AS_HELP_STRING([--with-openssl], [Build with OpenSSL support]))
-AS_IF([test "x$with_openssl" != "xno"],
+AS_IF([test "x$with_openssl" = "xyes"],
       [PKG_CHECK_MODULES([OpenSSL], [openssl >= 1.1], [have_openssl=yes], [have_openssl=no])],
       [have_openssl=no])
 AS_IF([test "x$have_openssl" = "xyes"], [
+    LDFLAGS="${LDFLAGS} -lssl"
```

Comment:
PKG_CHECK_MODULES should set macros OpenSSL_CFLAGS, OpenSSL_LIBS and OpenSSL_LDFLAGS (see for other dependencies like Argon2)

Reply:
Fixed at 3dcca38221cd02784a129b4782fcc8e90d3daeac